clusterctl/cmd/validate_cluster.go

```diff
@@ -34,9 +48,113 @@ var validateClusterCmd = &cobra.Command{
 }
 
 func init() {
+	validateClusterCmd.Flags().StringVarP(
+		&vco.ClusterName, "cluster-name", "", "",
+		"The name of the cluster to be validated. This can be omitted if there is only one cluster being managed in the given namespace.")
+	validateClusterCmd.Flags().StringVarP(
+		&vco.Namespace, "namespace", "n", v1.NamespaceDefault,
+		"If present, validate the cluster in the specified namespace.")
 	validateCmd.AddCommand(validateClusterCmd)
 }
 
 func RunValidateCluster() error {
-	return errors.NotImplementedError
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return fmt.Errorf("Failed to create client for talking to the apiserver: %v", err)
+	}
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("Failed to create client for talking to the apiserver: %v", err)
+	}
+
+	cluster, err := getClusterObject(client, vco.ClusterName, vco.Namespace)
+	if err != nil {
+		return err
+	}
+
+	err = validateClusterAPIObjects(client, cluster)
+	if err != nil {
+		return err
+	}
+
+	// TODO(wangzhen127): Also validate the cluster in addition to the cluster API objects.
+
+	glog.Infof("Pass. The cluster %q in namespace %q is validated.", cluster.Name, vco.Namespace)
```

Comment:
It's not clear what "Pass" means. I like the UI of unit tests where you can see each test that ran and the status of each one. We should shoot for something similar.

```
Checking foo bar... PASS
Checking bar foo... FAIL
  - Failure log: reason for failure
Checking baz baz... PASS
```

Reply:
I thought about the unit test style. Ideally, I wanted to do the following:
```
// Print "Checking foo bar...", but not newline
err := RunFooBar()
// Print "PASS or FAIL" on the same line as "Checking foo bar..." and then new line
```
But glog seems to always print newline. Do you know how to work this around? The reason of separating them in 2 prints is because RunFooBar() may take a long time and it is better to tell the user what it is doing.

Reply:
Use stdout instead?

Reply:
Done